### PR TITLE
[Live] Improve disconnect overlay reliability

### PR DESCRIPTION
**Pull request by orta**
Only remove the observer for the disconnect overlay when we're… removing the live view controller

### CODE REVIEW

**Comment by orta** [comment]
Previously it would stop observing the disconnect socket signal when you open an info panel, now it requires a full "I want to leave" to trigger the disconnect.

File: CHANGELOG.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ releases:
         - Uses server-defined increment strategy. - ash
         - Updates to bid-handling in history. - ash
         - Fixes jump bid ordering. - ash
-        - Parnter name shows on iPhone. - ash
+        - Parter name shows on iPhone. - ash
```
Comment (ashfurrow) [inline CHANGELOG.yml]: Partner*

Comment (orta) [inline CHANGELOG.yml]: hahaha

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
  </tbody>
</table>

Time | Class | Function |
| --- | ----- | ----- |247.5ms | [AuctionViewController.swift](/artsy/eigen/tree/orta-disconnect/Artsy/View_Controllers/Auction/AuctionViewController.swift#L183) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
203.1ms | [SwiftyJSON.swift](/artsy/eigen/tree/orta-disconnect/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
179.1ms | [WebSocket.swift](/artsy/eigen/tree/orta-disconnect/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
98.1ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/orta-disconnect/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L25) | func ar_presentBlurredOverlayWithTitle(title: String, subtitle: String, buttonState: BlurredStatusOverlayViewCloseButtonState = default)
94.6ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/orta-disconnect/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L31) | (closure)
80.3ms | [Observable.swift](/artsy/eigen/tree/orta-disconnect/Pods/Interstellar/Sources/Observable.swift#L71) | private final func nextTokenHash() -> Int
72.7ms | [AuctionLotMetadataStackScrollView.swift](/artsy/eigen/tree/orta-disconnect/Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift#L18) | required init(viewModel: LiveAuctionLotViewModelType, salesPerson: LiveAuctionsSalesPersonType, sideMargin: String)
72.5ms | [LiveAuctionsAdminViewController.swift](/artsy/eigen/tree/orta-disconnect/Artsy/View_Controllers/Live_Auctions/LiveAuctionsAdminViewController.swift#L22) | @objc override func viewDidLoad()
62.9ms | [LiveAuctionLotViewController.swift](/artsy/eigen/tree/orta-disconnect/Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift#L49) | @objc override func viewDidLoad()
59.0ms | [RefinementOptionsViewController+Private.swift](/artsy/eigen/tree/orta-disconnect/Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift#L63) | func stackView() -> ORStackView
55.5ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/orta-disconnect/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L40) | (closure)


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

**Pull request merged**